framework/widgets/ListView.php

```diff
@@ -74,7 +96,22 @@ public function renderItems()
         $keys = $this->dataProvider->getKeys();
         $rows = [];
         foreach (array_values($models) as $index => $model) {
-            $rows[] = $this->renderItem($model, $keys[$index], $index);
+            $key = $keys[$index];
+            if ($this->beforeRow !== null) {
+                $row = call_user_func($this->beforeRow, $model, $key, $index, $this);
+                if (!empty($row)) {
```

Comment:
Should be check for `null` or `false`.
Empty string added to the items set may be a desiarable behavior.